blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -195,35 +189,31 @@ namespace blueprint
             // Clear error state from previous js evals
             errorText.reset();
 
-            if (!bundleValid)
-            {
+            try {
                 // Register internal React.js backend rendering methods
                 registerNativeRenderingHooks();
 
                 if (beforeBundleEval)
                     beforeBundleEval(bundle);
-            }
 
-            // Set bundleValid prior to evaluate. This ensures dispatchEvent/dispatchViewEvent
-            // can be called during EcmascriptEngine::evaluate. This is required to allow things
-            // like triggering View "Measure" callbacks when laying out the initial component tree.
-            bundleValid = true;
-            auto result = engine.evaluate(bundle);
+                auto result = engine.evaluate(bundle);
 
-            bundleValid = result != EcmascriptEngine::EvaluationError;
+                if (afterBundleEval)
+                    afterBundleEval(bundle);
 
-            if (bundleValid && afterBundleEval)
-                afterBundleEval(bundle);
+                if (hotReloadEnabled)
```

Comment:
This hotReload block should probably be moved up above the try-catch block. It won't trigger and evaluation until the file actually changes. With this block inside the try catch block users cannot fixup their bundle if there is an error on initial load as the bundle will never be added to the watcher due to the new exception approach.

Reply:
i.e. 

```
        {
            JUCE_ASSERT_MESSAGE_THREAD

            // Clear error state from previous js evals
            errorText.reset();
            
            if (hotReloadEnabled)
            {
                jassert(bundleWatcher);

                if (!bundleWatcher->watching(bundle))
                    bundleWatcher->watch(bundle);
            }

            try {
                // Register internal React.js backend rendering methods
                registerNativeRenderingHooks();

                if (beforeBundleEval)
                    beforeBundleEval(bundle);

                auto result = engine.evaluate(bundle);

                if (afterBundleEval)
                    afterBundleEval(bundle);

                return result;
            } catch (const EcmascriptEngine::Error& err) {
                handleBundleError(err);
                return juce::var();
            }
        }
```

Reply:
Ah, of course, great catch